### PR TITLE
Add `react-css-modules.macro`

### PR DESCRIPTION
**Pull request by bohdanbirdie**
Thanks for collecting all those macro's in the single spot 👍

### CODE REVIEW

**Comment by bohdanbirdie** [comment]
@jgierer12 ping

**Comment by jgierer12** [comment]
Hi @bohdanbirdie !
Thanks for opening a PR and sorry about the late response :sweat_smile: 

I am a bit in two minds about your macro. On one hand, it's certainly a nice-to-have developer experience feature. However I'm not entirely sure if that is worth adding a bunch of additional code and function calls to the bundle.

Could you tell me a bit more about the performance implications of using this macro?

**Comment by bohdanbirdie** [comment]
@jgierer12 hi
Sure!
Basically, I was inspired by `react-css-modules` and then by `babel-plugin-react-css-modules`
But the first one is deprecated and the second one has issues with SCSS
First one was used on the project that I started working on and we needed to update that for many reasons (deprecation is one of them). The second one is using PostCSS and many features from SCSS just doesn't work well because it can't extract all the class names from it. Like `@each` operator ignored and etc. There are plantly of plugins for PostCSS to support that, but they are so-called `async` and babel plugin only work with `sync` plugins so I came up with this solution

In terms of performance:
- it only adds small file to the build which is a helper function
- a helper function is actually something that developer can implement anyway on its own, simply a selector that take a string and trying to get values from CSS hashed classnames map with those given classnames
- there is also two implementation - with and without memoization which will be good for performance as well

I personally think that it might have better/same performance as `react-css-modules` but probably a bit worse then `babel-plugin-react-css-modules` because `babel-plugin-react-css-modules` extract classnames from the CSS file during the build time and not a run time

File: readme.md [hunk 1]
```diff
@@ -87,6 +87,7 @@
 - [`reactive.macro`](https://github.com/yesmeck/reactive.macro): Reduce React boilerplate
 - [`@ts-delight/inject-display-name.macro`](https://github.com/ts-delight/inject-display-name.macro): Inject display name into dynamically constructed components
 - [`@ts-delight/fluent-react.macro`](https://www.npmjs.com/package/@ts-delight/fluent-react.macro): Fluent, Terse and Type-safe alternative to JSX for React
+- [`https://github.com/bohdanbirdie/react-css-modules.macro`](https://www.npmjs.com/package/react-css-modules.macro): Map CSS Modules to `styleName` property.
```
Comment (jgierer12) [inline readme.md]:
```suggestion
- [`react-css-modules.macro`](https://github.com/bohdanbirdie/react-css-modules.macro): Map CSS Modules to `styleName` property
```

**Comment by jgierer12** [comment]
Got it, when I first read through it it seemed a lot worse (performance wise) than it actually is. Maybe you'd want to add a performance section to the readme, similar to the `babel-plugin-react-css-modules` one.

**Review by jgierer12 — CHANGES_REQUESTED** [review]
Just a few small changes to match the other items in the list, then we can merge this.

**Comment by bohdanbirdie** [comment]
@jgierer12 good point, thanks!
Will update soon

**Review by jgierer12 — APPROVED** [review]

**Pull request merged**